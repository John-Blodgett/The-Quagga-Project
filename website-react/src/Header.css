#web-header-master {
    width: 100vw;
    height: 10vh;
    background-color: #0D3B66;
}

#web-header-logo {
    display: inline-block;
    
    position: inherit;
}

#web-header-title {
    display: inline-block;
    font-size: xx-large;
    font-weight: bold;
    color: white;
}

#zebra-background {
    display: table-cell;
    width: inherit;
    height: inherit;
    vertical-align: middle;
    background-image: url('./images/Zebra\ 5.png');
}

#web-cog-wheel {
    float: right;
    padding-right: 1.75vw;
    padding-top: 2vh;
}

#web-cog-wheel > img {
    width: 30px;
    height: 30px;
}