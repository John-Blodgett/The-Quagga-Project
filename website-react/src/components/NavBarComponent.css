#navbar-component {
    display: inline-block;
    width: 12vw;
    text-align: center;
    line-height: 4vh;
    margin: 0 auto;
}

#navbar-component:hover {
    background-color: lightgray;
}

#navbar-component:active {
    background-color: gray;
}