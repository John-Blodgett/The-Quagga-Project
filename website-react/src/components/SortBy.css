.sbcFrame {
    position: relative;
    width: 175px;
    height: 400px;
    display: block;
    background: #0D3B66;
    border: 1px solid #000000;
    box-sizing: border-box;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 5px;
}

.sbcFrame > div:first-child {
    font-family: Roboto;
    font-size: 20px;
    font-weight: 500;
    color: #F6F7FC;
}

.sbcFrame > div {
    margin-top: 2px;
    font-family: Roboto;
    font-size: 14px;
    font-weight: 400;
    color: #F6F7FC;
}

.sbcFrame > div > input {
    margin: 0px;
    width: 10px;
    height: 10px;
    float: right;
    margin-right: 2px;
}