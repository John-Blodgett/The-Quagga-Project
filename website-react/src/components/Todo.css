.todoFrame {
    position: absolute;
    display: grid;
    width: 400px;
    grid-template-rows: 75px;
    box-sizing: border-box;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    max-height: 600px;
}
.todoFrame > asgn {
    display: grid;
    grid-template-columns: 25% 20% 50% 5%;
    border: 1px solid #0D3B66;
    border-top: 1px solid #0D3B66;
    border-bottom: 1px solid #0D3B66;
    line-height: 50px;
    

}
.todoFrame > asgn:first-child {
    background-color: #F95837;
}
.todoFrame > asgn:nth-child(2) {
    background-color: #FF7153;
}


.todoName {
    font-family: Roboto;
    font-size: 20px;
    font-weight: 600;
    color: black;
    
    padding-left: 15px;
}

.todoDesc {
    font-family: Roboto;
    color: black;
    font-weight: 700;
    background-color: inherit;
    height: 28px;
}
.todoPointsPossible {
    font-family: Roboto;
    color: black;
    font-weight: 700;
    background-color: inherit;
    height: 28px;
}
