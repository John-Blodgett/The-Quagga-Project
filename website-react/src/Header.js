import React from 'react';
import './Header.css';
import logo from './images/Logo.png'

function Header() {
  return (
    <>
        <div id="web-header-master">
            <div id="zebra-background">
                <div id="web-header-logo">
                    <img src={logo} alt="Quagga Logo"></img>
                </div>
                <div id="web-header-title">
                    Quagga
                </div>
                <div id="web-cog-wheel">
                    <img src={"/cogWheel2.png"} alt="Cog Wheel"></img>
                </div>
            </div>
        </div>
    </>
  );
}

export default Header;
