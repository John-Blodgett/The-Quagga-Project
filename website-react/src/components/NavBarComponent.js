import React from 'react'
import './NavBarComponent.css'

function NavBarComponent({name}) {
    return (
        <>
            <div id="navbar-component">
                {name}
            </div>
        </>
    )
}

export default NavBarComponent;
