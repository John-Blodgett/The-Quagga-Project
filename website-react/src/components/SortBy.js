import React from 'react';
import './SortBy.css';

export default function SortBy() {
    return ( 
        <div class="sbcFrame">
            <div> Sort by:</div>
            <div>Due Date<input type="checkbox" id="sort1" name="sort1" value="DueDate" /></div>
            <div>Assignment Points<input type="checkbox" id="sort2" name="sort2" value="AssignmentPoints" /></div>
            <div>Class<input type="checkbox" id="sort3" name="sort3" value="Class" /> </div>
            <div>Custom<input type="checkbox" id="sort4" name="sort4" value="Custom" /> </div>

        </div>
    )
}
