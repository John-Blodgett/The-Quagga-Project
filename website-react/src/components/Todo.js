import React from 'react';
import './Todo.css';

export default function Todo() {
    var sampleAssignment = {
        'name': 'HW10',
        'desc': 'HW10 is blah blah blah',
        'pointsPossible': 10,
        'className': 'CSC309',
        'id' : 1
        };
    var sampleAssignment2 = {
        'name': 'HW20',
        'desc': 'HW20 Sucks',
        'pointsPossible': 30,
        'className': 'CSC309',
        'id' : 2
        };
    const sortedAssignemnts = [sampleAssignment, sampleAssignment2]

    return (
        <div className = 'todoFrame'>
            {sortedAssignemnts.map(assign =>{
            return <asgn key = {assign.id}>  
                    <div className = 'todoName'>{assign.name}</div>
                    <div className = 'todoClassName'>{assign.className}</div>
                    <div className = 'todoDesc'>{assign.desc}</div>
                    <div className = 'todoPointsPossible'>/{assign.pointsPossible}</div>
                </asgn>
            })}
       </div>
    )
}
