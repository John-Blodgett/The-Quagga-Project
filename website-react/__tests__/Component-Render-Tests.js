import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/Header';
import Todo from '../src/components/Todo'
import SortBy from '../src/components/SortBy'
import Navbar from '../src/Navbar';
import NavBarComponent from '../src/components/NavBarComponent';
import NavBarItems from '../src/Navbar';

test('renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(<Todo />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Navbar correctly', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Navbar correctly', () => {
  const tree = renderer.create(<SortBy />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Navbar correctly', () => {
  const tree = renderer.create(<NavBarComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Navbar items list correctly', () => {
  const tree = renderer.create(<NavBarItems />).toJSON();
  expect(tree).toMatchSnapshot();
});