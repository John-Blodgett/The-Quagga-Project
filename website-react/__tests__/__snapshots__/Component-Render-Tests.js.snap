// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div
  id="web-navbar-master"
>
  <div
    id="navbar-component"
  >
    FAQ
  </div>
  <div
    id="navbar-component"
  >
    Personal To-do
  </div>
  <div
    id="navbar-component"
  >
    School To-do
  </div>
  <div
    id="navbar-component"
  >
    Completed
  </div>
  <div
    id="navbar-component"
  >
    About Quagga
  </div>
</div>
`;

exports[`renders Navbar correctly 2`] = `
<div
  class="sbcFrame"
>
  <div>
     Sort by:
  </div>
  <div>
    Due Date
    <input
      id="sort1"
      name="sort1"
      type="checkbox"
      value="DueDate"
    />
  </div>
  <div>
    Assignment Points
    <input
      id="sort2"
      name="sort2"
      type="checkbox"
      value="AssignmentPoints"
    />
  </div>
  <div>
    Class
    <input
      id="sort3"
      name="sort3"
      type="checkbox"
      value="Class"
    />
     
  </div>
  <div>
    Custom
    <input
      id="sort4"
      name="sort4"
      type="checkbox"
      value="Custom"
    />
     
  </div>
</div>
`;

exports[`renders Navbar correctly 3`] = `
<div
  id="navbar-component"
/>
`;

exports[`renders Navbar items list correctly 1`] = `
<div
  id="web-navbar-master"
>
  <div
    id="navbar-component"
  >
    FAQ
  </div>
  <div
    id="navbar-component"
  >
    Personal To-do
  </div>
  <div
    id="navbar-component"
  >
    School To-do
  </div>
  <div
    id="navbar-component"
  >
    Completed
  </div>
  <div
    id="navbar-component"
  >
    About Quagga
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  id="web-header-master"
>
  <div
    id="zebra-background"
  >
    <div
      id="web-header-logo"
    >
      <img
        alt="Quagga Logo"
        src="/Logo.png"
      />
    </div>
    <div
      id="web-header-title"
    >
      Quagga
    </div>
    <div
      id="web-cog-wheel"
    >
      <img
        alt="Cog Wheel"
        src="/cogWheel2.png"
      />
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="todoFrame"
>
  <asgn>
    <div
      className="todoName"
    >
      HW10
    </div>
    <div
      className="todoClassName"
    >
      CSC309
    </div>
    <div
      className="todoDesc"
    >
      HW10 is blah blah blah
    </div>
    <div
      className="todoPointsPossible"
    >
      /
      10
    </div>
  </asgn>
  <asgn>
    <div
      className="todoName"
    >
      HW20
    </div>
    <div
      className="todoClassName"
    >
      CSC309
    </div>
    <div
      className="todoDesc"
    >
      HW20 Sucks
    </div>
    <div
      className="todoPointsPossible"
    >
      /
      30
    </div>
  </asgn>
</div>
`;
